core/preconditioner/jacobi.cpp

```diff
@@ -231,11 +232,20 @@ void Jacobi<ValueType, IndexType>::generate(const LinOp *system_matrix,
     using csr_type = matrix::Csr<ValueType, IndexType>;
     const auto exec = this->get_executor();
     if (parameters_.max_block_size == 1) {
-        auto diag = as<DiagonalExtractable<ValueType>>(system_matrix)
-                        ->extract_diagonal();
-        auto temp = gko::Array<ValueType>::view(
-            exec, system_matrix->get_size()[0], diag->get_values());
+        auto diag = share(as<DiagonalLinOpExtractable>(system_matrix)
+                              ->extract_diagonal_linop());
+        auto diag_vt =
+            ::gko::detail::temporary_conversion<matrix::Diagonal<ValueType>>::
```

Comment:
same here: Is the fully qualified name necessary?
```suggestion
            detail::temporary_conversion<matrix::Diagonal<ValueType>>::
```

Reply:
I need to put `gko::`, or I will get the error
` error: 'temporary_conversion' is not a member of 'gko::preconditioner::detail'`
it look for the function in `gko::preconditioner::detail` not `gko::detail`